src/charge_controller/dev/ControllerUI.cpp

```diff
@@ -0,0 +1,106 @@
+#include <charge_controller/dev/ControllerUI.hpp>
+
+ControllerUI::ControllerUI(DEV::Encoder& encoder, DEV::Button& encoderButton, LCDDisplay& display, ControllerModel& model) : model(model), encoder(encoder), encoderButton(encoderButton), display(display) {
+    encoder.setRangeAndPosition(model.getEncoderRange(ControllerModel::State::PAGE_SELECT), 0);
+}
+
+void ControllerUI::process() {
+    switch (model.getState()) {
+    case ControllerModel::State::PAGE_SELECT:
+        switch (encoder.getPosition()) {
+        case ControllerModel::Page::MAIN:
+            model.setPage(ControllerModel::Page::MAIN);
+            break;
+        case ControllerModel::Page::SETTINGS:
+            model.setPage(ControllerModel::Page::SETTINGS);
+            break;
+        default:
+            //THIS SHOULD NEVER HAPPEN
+            break;
+        }
+        if (encoderButton.debounce(DEBOUNCE_TIME) && model.getPage() == ControllerModel::Page::SETTINGS) {
+            setModelState(ControllerModel::State::SETTING_SELECT);
+        }
+        break;
+    case ControllerModel::State::SETTING_SELECT:
+        switch (encoder.getPosition()) {
+        case ControllerModel::SelectedSetting::VOLTAGE:
+            model.setSelectedSetting(ControllerModel::SelectedSetting::VOLTAGE);
+            break;
+        case ControllerModel::SelectedSetting::CURRENT:
+            model.setSelectedSetting(ControllerModel::SelectedSetting::CURRENT);
+            break;
+        case ControllerModel::SelectedSetting::SAVE:
+            model.setSelectedSetting(ControllerModel::SelectedSetting::SAVE);
+            break;
+        case ControllerModel::SelectedSetting::QUIT:
+            model.setSelectedSetting(ControllerModel::SelectedSetting::QUIT);
+            break;
+        case ControllerModel::SelectedSetting::RESET:
+            model.setSelectedSetting(ControllerModel::SelectedSetting::RESET);
+        default:
+            //THIS SHOULD NEVER HAPPEN!!!!
+            break;
+        }
+        if (encoderButton.debounce(DEBOUNCE_TIME)) {
+            switch (encoder.getPosition()) {
+            case ControllerModel::SelectedSetting::VOLTAGE:
+                setModelState(ControllerModel::State::VOLTAGE_SELECT);
+                break;
+            case ControllerModel::SelectedSetting::CURRENT:
+                setModelState(ControllerModel::State::CURRENT_SELECT);
+                break;
+            case ControllerModel::SelectedSetting::SAVE:
+                model.saveVoltageAndCurrent();
```

Comment:
Should this also change the state?

Reply:
nope, when it changes the state to CURRENT_SELECT it basically means they are editing the value for the current, which requires different logic